pkg/apis/core/validation/validation.go

```diff
@@ -1533,6 +1533,34 @@ func validateStorageOSPersistentVolumeSource(storageos *core.StorageOSPersistent
 	return allErrs
 }
 
+// ValidatePVSecretReference check whether provided SecretReference object is valid in terms of secret name and namespace.
+// This also take a flag to validate the secret name against DNSSubDomainformat (<253char) and DNS1123Label (<63char) format
+// as its required by some PV sources today, ex: CSI. This flag can help on racheting validation of secretRef field.
+func ValidatePVSecretReference(secretRef *core.SecretReference, allowDNSSubDomainFormat bool, fldPath *field.Path) field.ErrorList {
```

Comment:
this refactor is pretty similar to what was started in https://github.com/kubernetes/kubernetes/pull/103024

cc @roycaihw 

if you're intending to use this to start validating secret references that currently have no validation (like https://github.com/kubernetes/kubernetes/pull/108418#discussion_r821940157), you'll need to make every aspect of this optional (whether to require name, namespace, whether to validate name, namespace, whether to validate using allowDNSSubDomainFormat)